go/ephemeral/user_ek.go

```diff
@@ -277,25 +333,26 @@ func verifySigWithLatestPUK(ctx context.Context, g *libkb.GlobalContext, uid key
 		}
 		latestPUK = upak.Current.GetLatestPerUserKey()
 		if latestPUK == nil || !latestPUK.SigKID.Equal(signerKey.GetKID()) {
-			// The latest PUK still doesn't match the signing key after a
-			// forced reload. Bail out, and set the `wrongKID` flag.
-			latestPUKSigningKIDString := "<nil>"
-			if latestPUK != nil {
-				latestPUKSigningKIDString = fmt.Sprint(latestPUK.SigKID)
-			}
-			return nil, true, fmt.Errorf("userEK returned for PUK signing KID %s, but latest is %s",
-				signerKey.GetKID(), latestPUKSigningKIDString)
+			wrongKID = true
 		}
 	}
 
-	// If we didn't short circuit above, then the signing key is correct. Parse
-	// the JSON and return the result.
 	parsedStatement := keybase1.UserEkStatement{}
-	err = json.Unmarshal(payload, &parsedStatement)
-	if err != nil {
-		return nil, false, err
+	if err = json.Unmarshal(payload, &parsedStatement); err != nil {
+		return nil, wrongKID, err
+	}
```

Comment:
I wonder if we should parse the JSON higher up. It would be confusing if this returned a parse error, but then also the client skipped over that error because of the KID. Parse errors here should always be fatal I think.